components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -4836,6 +5074,210 @@ public void deleteApplication(Application application) throws APIManagementExcep
+
+        String deleteApplicationKeyQuery = SQLConstants.REMOVE_APPLICATION_LIST_FROM_APPLICATION_KEY_MAPPINGS_SQL
+                .replaceAll("_IDS_",
+                        String.join(",", Collections.nCopies(applicationIdList.length, "?")));
+
+        String deleteDomainAppQuery = SQLConstants.REMOVE_APPLICATION_FROM_DOMAIN_MAPPINGS_SQL;
+
+        String deleteApplicationQuery = SQLConstants.REMOVE_APPLICATION_LIST_FROM_APPLICATIONS_SQL
+                .replaceAll("_IDS_",
+                        String.join(",", Collections.nCopies(applicationIdList.length, "?")));
+
+        String deleteRegistrationEntry = SQLConstants.REMOVE_APPLICATION_LIST_FROM_APPLICATION_REGISTRATIONS_SQL
+                .replaceAll("_IDS_",
+                        String.join(",", Collections.nCopies(applicationIdList.length, "?")));
+
+        boolean transactionCompleted = true;
+        try {
+            connection = APIMgtDBUtil.getConnection();
+            connection.setAutoCommit(false);
+            prepStmt = connection.prepareStatement(getSubscriptionsQuery);
+
+            int index = 0;
+            for (int applicationId : applicationIdList) {
+                index++;
+                prepStmt.setInt(index, applicationId);
+            }
+
+            rs = prepStmt.executeQuery();
+
+            List<Integer> subscriptions = new ArrayList<Integer>();
+
+            while (rs.next()) {
+                subscriptions.add(rs.getInt("SUBSCRIPTION_ID"));
+            }
+
+            prepStmtGetConsumerKey = connection.prepareStatement(getConsumerKeyQuery);
+
+            int index = 0;
+            for (int applicationId : applicationIdList) {
+                index++;
+                prepStmtGetConsumerKey.setInt(index, applicationId);
+            }
+
+            rs = prepStmtGetConsumerKey.executeQuery();
+
+            deleteDomainApp = connection.prepareStatement(deleteDomainAppQuery);
+
+            while (rs.next()) {
+                String consumerKey = rs.getString(APIConstants.FIELD_CONSUMER_KEY);
+                String keyManagerName = rs.getString("NAME");
+                String keyManagerOrganization = rs.getString("ORGANIZATION");
+                // This is true when OAuth App has been created by pasting consumer key/secret in the screen.
+                String mode = rs.getString("CREATE_MODE");
+                if (consumerKey != null) {
+                    deleteDomainApp.setString(1, consumerKey);
+                    deleteDomainApp.addBatch();
+                    KeyManager keyManager =
+                            KeyManagerHolder.getKeyManagerInstance(keyManagerOrganization, keyManagerName);
+                    if (keyManager != null) {
+                        try {
+                            keyManager.deleteMappedApplication(consumerKey);
+                        } catch (APIManagementException e) {
+                            log.error("Error while Deleting Client Application", e);
```

Comment:
Any reason for only log if there is an error? shouldn't we hv to stop the flow? if we hv to continue u need to say why in a comment